pkg/agent/keyvalue_store.go

```diff
@@ -0,0 +1,17 @@
+package agent
+
+// Retriever is an interface for getting data from a persistent key/value store.
+type Retriever interface {
+	Get(key []byte) (value []byte, err error)
+}
+
+// Storer is an interface for setting data in a persistent key/value store.
+type Storer interface {
+	Set(key, value []byte) error
+}
+
+// RetrieverStorer is an interface that groups the Get and Set methods.
+type RetrieverStorer interface {
```

Comment:
`Getter` and  `Setter` maybe? To match the functions. Not really important though

Reply:
Yeah I've waffled back and forth on a half dozen naming schemes 😂 
Those names are a bit easier, I'll try it